### PR TITLE
Implement fft and ifft

### PR DESCRIPTION
**Pull request by ManoBharathi93**
2 Task is completed
- [x] fft
- [x]  ifft

### CODE REVIEW

**Comment by AnnaTz** [comment]
Hi! You will need to add tests for the added functions as well. Also, have you opened issues related to a ToDo list? If so, kindly pass the link here.

**Comment by AnnaTz** [comment]
You will need to follow our testing policy using the hypothesis module. There are many examples in our codebase like here https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py

**Comment by ManoBharathi93** [comment]
> You will need to follow our testing policy using the hypothesis module. There are many examples in our codebase, e.g. [here](https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py) We have a section in our docs explaining this [here](https://lets-unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html)

Thank you ! This code is working fine in local environment but couldn't pass tests ?~
```
import ivy
import jax.numpy as jnp
from hypothesis import given
import hypothesis.strategies as st

def ivy_fft(x):
    if ivy.backend.is_numpy(x):
        return jnp.fft.fft(x)
    return ivy.jax_backend.jax_fft(x)

def ivy_ifft(x):
    if ivy.backend.is_numpy(x):
        return jnp.fft.ifft(x)
    return ivy.jax_backend.jax_ifft(x)

@given(st.lists(st.complex_numbers()))
def test_ivy_fft(values):
    # convert to Ivy array
    x = ivy.array(values)
    # compute FFT using Ivy
    y = ivy_fft(x)
    # compute FFT using NumPy for comparison
    np_y = jnp.fft.fft(values)
    # assert equality
    assert ivy.to_numpy(y) == np_y

@given(st.lists(st.complex_numbers()))
def test_ivy_ifft(values):
    # convert to Ivy array
    x = ivy.array(values)
    # compute IFFT using Ivy
    y = ivy_ifft(x)
    # compute IFFT using NumPy for comparison
    np_y = jnp.fft.ifft(values)
    # assert equality
    assert ivy.to_numpy(y) == np_y
```

**Comment by AnnaTz** [comment]
> > You will need to follow our testing policy using the hypothesis module. There are many examples in our codebase, e.g. [here](https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py) We have a section in our docs explaining this [here](https://lets-unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html)
> 
> Thank you ! This code is working fine in local environment but couldn't pass tests ?~
> 
> ```
> import ivy
> import jax.numpy as jnp
> from hypothesis import given
> import hypothesis.strategies as st
> 
> def ivy_fft(x):
>     if ivy.backend.is_numpy(x):
>         return jnp.fft.fft(x)
>     return ivy.jax_backend.jax_fft(x)
> 
> def ivy_ifft(x):
>     if ivy.backend.is_numpy(x):
>         return jnp.fft.ifft(x)
>     return ivy.jax_backend.jax_ifft(x)
> 
> @given(st.lists(st.complex_numbers()))
> def test_ivy_fft(values):
>     # convert to Ivy array
>     x = ivy.array(values)
>     # compute FFT using Ivy
>     y = ivy_fft(x)
>     # compute FFT using NumPy for comparison
>     np_y = jnp.fft.fft(values)
>     # assert equality
>     assert ivy.to_numpy(y) == np_y
> 
> @given(st.lists(st.complex_numbers()))
> def test_ivy_ifft(values):
>     # convert to Ivy array
>     x = ivy.array(values)
>     # compute IFFT using Ivy
>     y = ivy_ifft(x)
>     # compute IFFT using NumPy for comparison
>     np_y = jnp.fft.ifft(values)
>     # assert equality
>     assert ivy.to_numpy(y) == np_y
> ```

Please read our testing documentation that I linked above. There is also an abundance of existing functions and tests you can use as reference.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
